core/src/main/scala/org/apache/spark/sql/crossdata/catalog/XDCatalog.scala

```diff
@@ -55,6 +55,11 @@ abstract class XDCatalog(val conf: CatalystConf = new SimpleCatalystConf(true),
     }
   }
 
+  def tableExistsInCatalog(tableIdentifier: Seq[String]): Boolean = {
+    val (table, database) = tableIdToTuple(tableIdentifier)
+    lookupTable(table, database).fold(false)(crossdataTable => true)
```

Comment:
It's ok as it is but you might like this one:

`lookupTable.tupled(tableIdToTuple(tableIdentifier))`